topics/ecology/tutorials/x-array-map-plot/tutorial.md

```diff
@@ -0,0 +1,394 @@
+>    - The second file `info file` provide a summary of the **Xarray Dataset** contained in your netCDF file.
+{: .hands_on}
+## Map Plotting using **NetCDF xarray map plotting**
+
+> ### {% icon hands_on %} Hands-on: Map Plotting
+> The air temperatures in the last two hours of the day are plotted here : 
+>
+> 1. {% tool [NetCDF xarray map plotting](toolshed.g2.bx.psu.edu/repos/ecology/xarray_mapplot/xarray_mapplot/0.20.2+galaxy0) %} with the following parameters:
+>    - {% icon param-file %}
+*"Netcdf file"*: `outfile_17.nc`
+>    - {% icon param-file %} *"Tabular of variables"*: `Metadata infos from outfile_17.nc` (output of **NetCDF xarray Metadata Info** >{% icon tool %})
+>    - *"Choose the variable to plot"*: `air_temperature_at_2_metres`
+>    - *"Name of latitude coordinate"*: `lat`
+>    - *"Name of longitude coordinate"*: `lon`
+>    - *"Datetime selection"*: `No`
+>    - *"Range of values for plotting e.g. minimum value and maximum value (minval,maxval) (optional)"*: `220,320`
+>    - *"Add country borders with alpha value [0-1] (optional)"*: `1.0`
+>    - *"Add coastline with alpha value [0-1] (optional)"*: `1.0`
+>    - *"Add ocean with alpha value [0-1] (optional)"*: `1.0`
+>    - *"Specify plot title (optional)"*: `Projection :  Mercator 17:00 UTC `
+>    - *"Specify which colormap to use for plotting (optional)"*: `lajolla`
+>    - *"Specify the projection (proj4) on which we draw e.g. {"proj":"PlateCarree"} with double quote (optional)"*: `{'proj': 'Mercator', 'central_longitude': 12.0}`
+>
+>![ECMWF Reanalysis Air temperature a 2 metres on 2022-05-25 at 17:00:00 UTC](../../images/ECMWF_Reanalysis_17.png)
+>
+>    > ### {% icon comment %} Why shifting longitudes?
+>    >
+>    > Longitudes are measured in degrees from 0 to 360. We need to change longitudes so that `NetCDF xarray map plotting` can properly plot our dataset because we don't have worldwide data but only cover Europe.
+>    {: .comment}
+>
+> 
+{: .hands_on}
+
+
+
+> ### {% icon question %} Questions
+>
+> 1. What are the different kinds of projections and colormaps that can be used?
+>
+> > ### {% icon solution %} Solution
+> >
+> > 1.There are many projections which can be used in the `NetCDF xarray map plotting` tool. Find here a list of valid projections. Different projections have different purposes and need to be carefully chosen. Follow this sheet to [find the syntax for projections](https://github.com/Quickbeasts51429/xarray_projection). 
+> > 
+>> When it comes to conveying the correct information, through visualisation, colors play a major role. Suppose you want to display a cold region, its an obvious practice of using cooler tones such as a `blue`. Thus it is important to understand the choices we have. Find the [complete documentation of ColorMaps available here](https://github.com/Quickbeasts51429/Xarray_ColorMaps/blob/main/index.md)
```

Comment:
Same for here. It would be nice to find a nice way to get this information in the tutorial itself with guidance on what to choose and for which purposes.

Reply:
Oh Yeah, that can definitely be done...But the count of projections and colormaps are too much, I can include the steps for plotting using one each of the two, i.e. the colormaps and projections and then embed a link to the reference documentations. Would that be alright?

Reply:
Or may be , I should show some montages of colormaps and projections?
This could include more maps in smaller space. How does this sounds?

Reply:
Very good suggestion!

Reply:
Showing in priority projections & colormaps that are the most popular.

Reply:
> 

Oh, that's nice ....then I'll be following this and will plotting priority projections and most used colormaps. Just one another doubt, do I have to start from scratch with a new workflow or making the changes in this tutorial would be okay? I believe the workflow.ga file would be insufficient in content if I edit this same tutorial. Please let me what works the best.

Reply:
Make the changes in the tutorial. We will update the workflow later.